docs/guides/nrfconnect_factory_data_configuration.md

```diff
@@ -0,0 +1,291 @@
+-   [Programming factory data](#programming-factory-data)
+
+<hr>
+
+<a name="overview"></a>
+
+## Overview
+
+The factory data parameter set includes information such as a manufacturer name, date of manufacturing, product certifications, and PCB versioning. 
+
+For the nRF Connect platform, factory data is stored by default in the internal Flash memory in a separate partition. This helps to keep factory partition secure by applying hardware protection. Parameters are read at the boot time of a device, and then they can be used in the Matter stack and user application.
+
+> Note: Encryption and security of the factory data partition is not provided yet for this feature.
+
+### Factory data components
+
+A factory data set consists of following parameters:
+
+|Key name|Full name|Length|Format|Conformance|Description|
+|:------:|:-------:|:----:|:----:|:---------:|:---------:|
+|**`sn`**|`serial number`|<1, 32> B|*ASCII string*|mandatory|A serial number parameter defines an unique number of manufactured device. Maximum length of serial number is 32 characters.|
+|**`vendor_id`**|`vendor ID`|2 B|*uint16*|mandatory|A MATTER-assigned id for the organization responsible for producing the device.|
+|**`product_id`**|`product ID`|2 B|*uint16*|mandatory|A unique id assigned by the device vendor to identify the product or device type. Defaults to a MATTER-assigned id designating a non-production or test "product".|
+|**`vendor_name`**|`vendor name`|<1, 32> B|*ASCII string*|mandatory|A human-readable vendor name which provides a simple string containing identification of device's vendor for the application and Matter stack purposes.|
+|**`product_name`**|`product name`|<1, 32> B|*ASCII string*|mandatory|A human-readable product name which provides a simple string containing identification of the product for the  the application and Matter stack purposes.|
+|**`date`**|`manufacturing date`|<8, 10> B|*ISO 8601*|mandatory|A manufacturing date specifies the date that the device was manufactured. The format used for providing a manufacturing date is ISO 8601 e.g. YYYY-MM-DD.|
+|**`hw_ver`**|`hardware version`|2 B|*uint16*|mandatory|A hardware version number specifies the version number of the hardware of the device. The meaning of its value, and the versioning scheme, are vendor defined.|
+|**`hw_ver_str`**|`hardware version string`|<1, 64> B|*uint16*|mandatory|A hardware version string parameter specifies the version of the hardware of the device as a more user-friendly value than that presented by the hardware version integer value. The meaning of its value, and the versioning scheme, are vendor defined.|
+|**`rd_uid`**|`rotating device ID unique ID`|<20, 36> B|*hex string*|mandatory|The unique ID for rotating device ID consists of a randomly-generated 128-bit or longer octet string. This parameter should be protected against reading or writing over the air after initial introduction into the device, and stay fixed during the lifetime of the device.|
```

Comment:
I wonder if we should use *hex string* as a type. On the device it's stored as *byte string* and is only encoded as *hex string* in a JSON file (due to JSON format limitations).

Reply:
Hmm, ok If we decided to inform users about the possibility of creating their factory data accessor, we should put here types that describe values in the final hex file.